vizro-core/src/vizro/models/_navigation/_navigation_utils.py

```diff
@@ -0,0 +1,21 @@
+import itertools
+
+from vizro.managers import model_manager
+
+
+def _validate_pages(pages):
+    """Validator for reuse in `Navigation` and `Accordion` to validate pages."""
+    from vizro.models import Page
+
+    pages_as_list = list(itertools.chain(*pages.values())) if isinstance(pages, dict) else pages
```

Comment:
Do we actually need this conversion? Since this validator always runs after the one that coerces to dictionary. The pydantic type hint `Dict[str, List[str]]` should be guaranteed by this point?

Reply:
Ah yes, you are right 👍 - I've removed the list conversion as we can loop through the` itertools.chain `object directly

Reply:
EDIT: We actually do need it - saw one of the unit tests failing just now 😄 We need the list conversion so the next check properly catches the case if `navigation.pages = {"SELECT PAGE": []}`

So we need it for this check after:

```
if not pages_as_list:
    raise ValueError("Ensure this value has at least 1 item.")
```

Otherwise this unit tests fails:
<img width="739" alt="Screenshot 2023-10-18 at 10 10 56" src="https://github.com/mckinsey/vizro/assets/90609403/073abf1c-b4a8-410f-bf98-d49f9256c6be">

Reply:
Sorry, I wasn't at all clear in my original comment. What I meant is not whether we need to do the coercion to list but whether we need to do the `isinstance` check? Can't we just do this because we're guaranteed that `pages` is `Dict[str, List[str]]` by this point?

```
pages = list(itertools.chain(*pages.values()))
```

Reply:
Ah got it, it works for the Accordion, but not for the Navigation. In the Accordion pages will always be in a Dict format when it hits this validator, but this is not the case for the Navigation e.g. you can provide something like this: Navigation(pages=["Page 1", "Page 2", "Page 3"]).

It relates to the question yesterday when I was wondering whether we should move the dict conversion to the Navigation as well, but I think we agreed that we would only have that in the Accordion as as soon as the NavBar comes in we would require a different dict format than what the Accordion requires.

Reply:
Ah yes, makes complete sense, thanks for the explanation 👍 I forgot that the same validator is used by `Navigation` where we can still have `List[str]`.

It's great that the tests are catching all of these cases so we don't make a mistake based on my bad suggestions 😀  💯